leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/ArgumentParser.java

```diff
@@ -0,0 +1,5 @@
+package org.eclipse.leshan.core.request.execute;
+
+public interface ArgumentParser {
```

Comment:
> I also created parser/serializer similar to Link solution. I'm not sure, maybe it's overkill for such simple format.

You maybe right about this is maybe overkill. I haven't strong opinion. 
If you simplify it what does it looks like ? 
- just removing the Interface class ? 
- moving the serialization/deserialization in  `Arguments` class ?

Reply:
I try to follow by clean code rules. For example single-responsibility principle: parsing, serializing and holding arguments are different responsibilities. Introducing interfaces also helps with decoupling the code.

But that's only the theory - in practice often clean code means more code. Different developers have different opinions about how to organize the code, so I just try to balance.

In project we have for example Link with parsers, and LwM2mPath without parser. For LwM2mPath format is very simple, so creating parsers/serializers would be overkill. For Link the format is more complicated and by introducing parser we allow to customize the rules.

I think the arguments have quite simple format, so maybe creating parsers and serializers would introduce unnecessary noise to the code.